tests-dev/create_log.py

```diff
@@ -0,0 +1,204 @@
+                    for test in val:
+                        case, config = get_testcase(test)
+                        machine_check = machine_check_off(MACHINE_ID, config)
+                        if machine_check:
+                            JOB_NR+=1
+                            TEST_NAME = case
+                            TEST_ID   = TEST_NAME+'_'+RT_COMPILER
+                            TEST_LOG  = 'rt_'+TEST_ID+'.log'
+                            TEST_LOG_TIME= 'run_'+TEST_ID+'_timestamp.txt'
+                            if 'dependency' in config.keys():
+                                DEP_RUN = str(config['dependency'])+'_'+RT_COMPILER
+                            else:
+                                DEP_RUN = ""
+                            PASS_CHECK = 'Test '+TEST_ID+' PASS'
+                            MAXS_CHECK = 'The maximum resident set size (KB)'
+                            pass_flag = False
+                            with open('./logs/log_'+MACHINE_ID+'/'+TEST_LOG) as f:
+                                if PASS_CHECK in f.read():
+                                    pass_flag = True
+                            f.close()
+                            if pass_flag:
+                                f = open('./logs/log_'+MACHINE_ID+'/'+TEST_LOG_TIME)
+                                timing_data = f.read()
+                                first_line = timing_data.split('\n', 1)[0]
+                                etime = str(int(first_line.split(",")[4].strip()) - int(first_line.split(",")[1].strip()))
+                                rtime = str(int(first_line.split(",")[3].strip()) - int(first_line.split(",")[2].strip()))
+                                etime_min, etime_sec = divmod(int(etime), 60)
+                                etime_min = f"{etime_min:02}"; etime_sec = f"{etime_sec:02}"
+                                rtime_min, rtime_sec = divmod(int(rtime), 60)
+                                rtime_min = f"{rtime_min:02}"; rtime_sec = f"{rtime_sec:02}"
+                                time_log = " ["+etime_min+':'+etime_sec+', '+rtime_min+':'+rtime_sec+"]"
+                                f.close()
+                            with open('./logs/log_'+MACHINE_ID+'/'+TEST_LOG) as f:
+                                if pass_flag :
+                                    rtlog_file = f.readlines()
+                                    for line in rtlog_file:
+                                        if MAXS_CHECK in line:
+                                            memsize= line.split('=')[1].strip()
+                                    test_log = 'PASS -- TEST '+TEST_ID+time_log+' ('+memsize+' MB)\n'
+                                    PASS_NR += 1
+                                else:
+                                    test_log = 'FAIL -- TEST '+TEST_ID+'\n'
+                                    failed_list.append(TEST_NAME+' '+RT_COMPILER)
```

Comment:
Why did test fail? e.g., TEST_RESULT in rt.sh

Reply:
I agree we can further decompose 'run_'+TEST_ID+'.log' for a root cause analysis of failed cases. Maybe we can create another after we merge this pr.  That feature can be combined when we add a vertical hierarchical test key.

Reply:
Isn't this supposed to match the behavior of current rt.sh? A test failure should report why it failed (e.g., https://github.com/ufs-community/ufs-weather-model/issues/2330). This is separate from a hierarchical test key